### ISSUE TITLE
cant load other swatch's

### ISSUE THREAD

**Issue by shane3140** [trevjonez/Pantone-Color-Book-Builder#2]
would it be possible to change the script to just load the swatch's that are open in the main swatch tab. it's easy just to add the swatches from other swatch libraries into the main swatch tab.

**Comment by trevjonez** [comment]
I don't currently have an active subscription. but really the thing that determines what colors go in is the array of strings on line one. you might play with just updating that as you see fit.

**Comment by shane3140** [comment]
Hi mate, I don’t know anything about scripts.

**Comment by trevjonez** [comment]
this would actually be a great first introduction to doing something with them.
the script is just a text file nothing you can mess up. 

if you change the top line it will insert different colors based on what is there.
`var colors = new Array("PANTONE Yellow C", "PANTONE Yellow 012 C") ` 
would make it only do those two swatch colors. 
if you add ` "NextPage" ` inside the list of values it will make it start the next page before continuing.

**Comment by shane3140** [comment]
Hi Trevor, I did open the file in notepad. I seen you has listed all the pantone colours. I thought if you took all the colour names out and and put a script into use the main swatch lib. It might work. I might have a look at it over this week end.
Thanks shane

**Comment by schroef** [comment]
yes that is possible, i dont quite understand why this script uses a baked in array of the Pantone Names. If you simply use ```app.activeDocument.swatches.getSelected()``` It will return an array of all the selected swatches, then you can easily extract the names and the colors.

@shane3140 

have a look at the script "Render Swatch Legends", i think this is what you need. Have a look at my latest post, ive adjusted it so it also makes HEX and will ask you what color info you want to show.
https://community.adobe.com/t5/illustrator-discussions/illustrator-javascript-render-swatch-legend-lab-colour-values-incorrect/m-p/13094886#M329919

Here's a simple loop which gets the selected swatches and returns the names. Therefor you dont need array with all the names and you can use it on any document you want
```
var doc = app.activeDocument;
var swatches = doc.swatches.getSelected()
for (i=0;i<swatches.length; i++){
    alert(swatches[i].name)
}
```
If you leave out ```.getSelected()```. by default it will return all the swatches available in your document

**Comment by schroef** [comment]
Ive updated the script, its smaller in size due to the hardcoded array being deleted. One small thing i do see, i doesnt add or adjust the artboards. So all the pages are being outside any printable area

![doesnt use artboards](https://user-images.githubusercontent.com/6923008/202494945-250147e5-ba59-452d-9c82-a843f029a1d5.png)


Here's an example i ran with added swatches. You do need to add you artwork or swatches to the document and clear all those added PMS colors from the original document
![Pantone Book Maker-Updated](https://user-images.githubusercontent.com/6923008/202495155-bf7f50ad-f61d-4d00-bb14-b1dea46d6bed.png)

Here's the scripts
```
var swatches = app.activeDocument.swatches.getSelected();
var colors = [];
for(i=0;i<swatches.length; i++){
    colors.push(swatches[i].name);
}
var pages = Math.ceil(colors.length/7);
var Xoffset = 144;
var Yoffset = -684;
var lineLength = 19;

function makePages(){
    var cPage = 1;
    var shiftX = Xoffset * cPage;
    var shiftY = 0;
    var X = 1;
    var Y = 0;
    var pageOne = app.activeDocument.layers.getByName("PAGE 1");
    while(cPage <= pages){
        
        if(X >= lineLength){
                X = 0;
                Y++;
        }
        cPage++;
        var nextPage = app.activeDocument.layers.add();
        nextPage.name = "PAGE " + cPage.toString();
        var content = pageOne.pageItems.getByName("LAYOUT");
        var newContent = content.duplicate();
        newContent.move(nextPage, ElementPlacement.INSIDE);
        newContent.textFrames.getByName("PAGE").contents = cPage.toString();
        newContent.translate(Xoffset*X,Yoffset*Y);
        X++;
    }
}

function fillPages(){
    var pCount = 1;
    var cCount = 0;
    for(var i = 1; i <= pages; i++){
            var cPage = app.activeDocument.layers.getByName("PAGE " + pCount.toString());
            var layout = cPage.groupItems.getByName("LAYOUT");
            for(var j = 1; j < 8; j++){
                    var cBlock = layout.groupItems.getByName(j.toString());
                    if(colors[cCount]=="NextPage"){
                        cCount++;
                        break;
                    }
                    cBlock.textFrames.getByName("COLOR").contents = colors[cCount].replace("PANTONE ", "");
                    cBlock.pathItems.getByName("FILL").fillColor = app.activeDocument.swatches.getByName(colors[cCount]).color;
                    cCount++;
                    if(cCount >= colors.length){
                        return;
                    }
            }
            pCount++;
    }
}

makePages();
fillPages();
```

**Comment by trevjonez** [comment]
> i dont quite understand why this script uses a baked in array of the Pantone Names.

Its been a long time since I wrote or even used the scripts but if I recall correctly I did the hardcoded array  so that the script would layout the pages matching the real pantone books I was using at the time. Comparing printer profile on whatever material I was using vs the real pantone books was easier that way.

**Comment by schroef** [comment]
How did you actually printed this, because i noticed that there are no extra artboards created. So on export it will export only one page. Unless you manually adjust that later

**Comment by trevjonez** [comment]
> How did you actually printed this, because i noticed that there are no extra artboards created. So on export it will export only one page. Unless you manually adjust that later

I believe I just adjusted manually for print and to add bounding marks for cutting either on our laser or on a graphtec machine.

**Comment by schroef** [comment]
Aha i see. I was thinking about adding code, so it will add artboards. That would be much easier
